### ISSUE TITLE
[patch-package fix] #2979 label covers text on iOS multiline TextInput

### ISSUE THREAD

**Issue by Svarto** [callstack/react-native-paper#3020]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `react-native-paper@4.11.1` for the project I'm working on.

There was a PR that tried to fix the iOS multiline label covering the underlying text, unfortunately this added a bigger issue of the label covering the underlying text even if no label was present. On top of this, it used a backgroundColor that was not controllable from the component (hardcoded in react-native-paper) - which meant if someone specified their own backgroundColor the label would have a different backgroundColor.

This simple patch-package checks if a label is present, and if so renders the label container, and also matches the backgroundColor with the backgroundColor set in style prop of TextInput.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/react-native-paper/src/components/TextInput/TextInputFlat.tsx b/node_modules/react-native-paper/src/components/TextInput/TextInputFlat.tsx
index d5c9bee..8700a07 100644
--- a/node_modules/react-native-paper/src/components/TextInput/TextInputFlat.tsx
+++ b/node_modules/react-native-paper/src/components/TextInput/TextInputFlat.tsx
@@ -331,7 +331,7 @@ class TextInputFlat extends React.Component<ChildTextInputProps> {
             },
           ]}
         >
-          {!isAndroid && multiline && (
+          {!isAndroid && multiline && label && (
             // Workaround for: https://github.com/callstack/react-native-paper/issues/2799
             // Patch for a multiline TextInput with fixed height, which allow to avoid covering input label with its value.
             <View
@@ -341,7 +341,7 @@ class TextInputFlat extends React.Component<ChildTextInputProps> {
                 StyleSheet.absoluteFill,
                 dense ? styles.densePatchContainer : styles.patchContainer,
                 {
-                  backgroundColor: containerStyle.backgroundColor,
+                  backgroundColor: viewStyle.backgroundColor,
                   left: paddingLeft,
                   right: paddingRight,
                 },
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by lukewalczak** [comment]
Hey @Svarto, I will apply & run your patch to check if everything is correct. If it will be ok, are you willing to open the PR?

**Comment by lukewalczak** [comment]
Hello @Svarto, please open the PR with your approach with one additional change:

```
backgroundColor: viewStyle.backgroundColor || containerStyle.backgroundColor
```

**Comment by jensonjohn001** [comment]
Facing same issue.. Please  open the PR @Svarto 

![simulator_screenshot_61719386-2533-4E6C-B9E5-4CBCF7255A3F](https://user-images.githubusercontent.com/35399770/149178922-2f8ab999-21ff-465c-96e9-a663adc7beae.png)

**Comment by Insider32** [comment]
Waiting for that PR too

**Comment by Svarto** [comment]
@Insider32 @jensonjohn001  alright, I am on it!

**Comment by VictorioMolina** [comment]
Any updated on this? I am using `dense={true}` and `multiline={true}`, and with no label, the label container covers my text.
   
"react-native": "https://github.com/expo/react-native/archive/sdk-44.0.0.tar.gz",
 "react-native-paper": "^4.11.1",

**Comment by lukewalczak** [comment]
> Any updated on this? I am using dense={true}, mode="flat", and multiline={true}, and without using any label, its container covers my text.

The PR is ready, waiting for fixing lint. In the meantime you can use the patch which @Svarto provided.

**Comment by lukewalczak** [comment]
Patch is released in `4.11.2`

**Comment by NotoriousGBG** [comment]
@lukewalczak I just upgraded to 4.11.2 but the issue persists.
